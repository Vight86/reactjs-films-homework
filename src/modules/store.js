import {
  createStore, applyMiddleware, combineReducers,
} from 'redux';
import camelMiddleware from 'redux-camel';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';

import moviesReducer from './movies/index';
import queryReducer from './query/index';
import genresReducer from './genres/index';
import statusReducer from './status/index';
import isGridReducer from './isGrid/index';

const createRootReducer = (history) => combineReducers({
  router: connectRouter(history),
  query: queryReducer,
  movies: moviesReducer,
  genres: genresReducer,
  status: statusReducer,
  isGrid: isGridReducer,
});

export const history = createBrowserHistory();

const composeEnhancers = composeWithDevTools({ trace: true, traceLimit: 25 });

const persistedState = localStorage.getItem('movieDBState');
const preloadedState = JSON.parse(persistedState) || {};

const store = createStore(
  createRootReducer(history),
  preloadedState,
  composeEnhancers(
    applyMiddleware(
      routerMiddleware(history),
      thunk,
      camelMiddleware(),
    ),
  ),
);

export default store;
