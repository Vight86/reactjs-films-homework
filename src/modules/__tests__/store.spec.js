import store from '../store';

describe('testing redux store', () => {
  it('redux store is setup correctly', () => {
    expect(store.getState()).toMatchSnapshot();
  });
});
