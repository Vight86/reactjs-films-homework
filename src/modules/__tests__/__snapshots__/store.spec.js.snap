// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing redux store redux store is setup correctly 1`] = `
Object {
  "filters": Object {
    "genreId": 0,
  },
  "genres": Array [],
  "movies": Object {
    "currentMovieTrailerKey": "",
    "movies": Array [],
    "page": 1,
    "pageSize": 16,
    "totalPages": 0,
  },
  "query": Object {
    "adult": false,
    "apiKey": "e5dc29ce2b6d31bb5c5306f6a7469ba2",
    "apiPath": "/movie/popular",
    "basePath": "https://api.themoviedb.org/3",
    "genresPath": "/genre/movie/list",
    "lang": "en-US",
    "searchQuery": "",
  },
  "status": Object {
    "status": "loading",
  },
}
`;
